cfme/cloud/keypairs.py

```diff
@@ -185,6 +189,24 @@ def download_private_key(self):
         view.toolbar.configuration.item_select('Download private key')
         view.flash.assert_no_error()
 
+    @property
+    def exists(self):
+        try:
+            wait_for(lambda: navigate_to(self, "Details"), num_sec=10, delay=2,
```

Comment:
I think this should only check entity exists or not in the model page.
You can wait for the entity in the test. Maybe you are facing this problem with `create`?
If yes I think you can take an extra argument in the `create` method and wait for entity create there.

Reply:
Fixed by adding wait_for to create method.